django/middleware/cache.py

```diff
@@ -121,6 +121,12 @@ def process_response(self, request, response):
             cache_key = learn_cache_key(
                 request, response, timeout, self.key_prefix, cache=self.cache
             )
+            # check header for "x-successive-cache" to avoid multiple cache sets
+            # if not found, set it as key used
+            if response.has_header("x-successive-cache"):
+                return response
+            response["x-successive-cache"] = cache_key
```

Comment:
I think maybe `X-Cache-Success` would be better (I think headers tend to be title cased), I have no idea how to tell whether it is unlikely to collide with other headers, I agree it _feels_ unlikely

We probably want to set the header as something short and constant (as we currently don't care about the value) maybe something like `"true"`

Reply:
after a bit more digging found that "Last-Modified" header will be present on cached responses after processing... If that exists we can confidently skip.